Service.Tests/ServiceProxyTest.cs

```diff
@@ -0,0 +1,277 @@
+                        .WithBody(SUCCESS_RESPONSE_BODY)
+                );
+
+            var response = await _sut.Put("/foo", "", body);
+            var statusCode = (int) response.StatusCode;
+            var responseBody = await response.Content.ReadAsStringAsync();
+
+            Check.That(statusCode).IsEqualTo(SUCCESS_STATUS_CODE);
+            Check.That(responseBody).IsEqualTo(SUCCESS_RESPONSE_BODY);
+        }
+
+        [Test]
+        public async Task TestPutWithQueryAndOptions()
+        {
+            var port = _server.Ports.First();
+            var initServiceOptions = new InitServiceOptions(port);
+            _sut = new ServiceProxy("localhost", initServiceOptions);
+            const string body = @"{ ""foo"": ""bar"" }";
+
+            _server
+                .Given(Request.Create().WithPath("/foo").WithParam("bar", "baz").WithBody(body).UsingPut())
+                .RespondWith(
+                    Response.Create()
+                        .WithStatusCode(SUCCESS_STATUS_CODE)
+                        .WithBody(SUCCESS_RESPONSE_BODY)
+                );
+
+            var headers = new Dictionary<string, string> {{"foo", "bar"}};
+            var response = await _sut.Put("/foo", "bar=baz", body, new ServiceOptions(port, Protocol.Http, headers));
+            var statusCode = (int) response.StatusCode;
+            var responseBody = await response.Content.ReadAsStringAsync();
+
+            Check.That(response.RequestMessage.Headers.GetValues("foo").First()).IsEqualTo("bar");
+            Check.That(statusCode).IsEqualTo(SUCCESS_STATUS_CODE);
+            Check.That(responseBody).IsEqualTo(SUCCESS_RESPONSE_BODY);
+        }
+        
+        [Test]
+        public async Task TestPatch()
+        {
+            var initServiceOptions = new InitServiceOptions(_server.Ports.First());
+            _sut = new ServiceProxy("localhost", initServiceOptions);
+            const string body = @"{ ""foo"": ""bar"" }";
+
+            _server
+                .Given(Request.Create().WithPath("/foo").WithBody(body).UsingPatch())
+                .RespondWith(
+                    Response.Create()
+                        .WithStatusCode(SUCCESS_STATUS_CODE)
+                        .WithBody(SUCCESS_RESPONSE_BODY)
+                );
+
+            var response = await _sut.Patch("/foo", "", body);
+            var statusCode = (int) response.StatusCode;
+            var responseBody = await response.Content.ReadAsStringAsync();
+
+            Check.That(statusCode).IsEqualTo(SUCCESS_STATUS_CODE);
+            Check.That(responseBody).IsEqualTo(SUCCESS_RESPONSE_BODY);
+        }
+
+        [Test]
+        public async Task TestPatchWithQueryAndOptions()
+        {
+            var port = _server.Ports.First();
+            var initServiceOptions = new InitServiceOptions(port);
+            _sut = new ServiceProxy("localhost", initServiceOptions);
+            const string body = @"{ ""foo"": ""bar"" }";
```

Comment:
Why the double quotes around `foo` and `bar`?

Reply:
It's the way to escape double quotes in verbatim string literals (you define them with a `@` at the beginning)